core/lib/resources/presence/presence_manager.js

```diff
@@ -119,11 +121,18 @@ PresenceManager.prototype.stampExpiration = function(message) {
 };
 
 PresenceManager.prototype.addClient = function(socketId, userId, userType,
-                                                      userData, callback) {
+                                                      userData, data, callback) {
+
+  // FIXME: Method signature is getting unmanageable.  
```

Comment:
Completely agree. Fix this. I don't care how anymore. I will take the review effort.